p2panda-rs/src/schema/schema.rs

```diff
@@ -60,6 +63,48 @@ impl Schema {
     pub fn as_cddl(&self) -> String {
         generate_cddl_definition(&self.fields)
     }
+
+    /// Access the schema's [`SchemaId`].
+    #[allow(unused)]
+    pub fn id(&self) -> &SchemaId {
+        &self.id
+    }
+
+    /// Access the schema view id.
+    #[allow(unused)]
+    pub fn view_id(&self) -> DocumentViewId {
+        let view_id = match self.id.clone() {
+            SchemaId::Application(_, view_id) => Some(view_id),
+            _ => None,
+        };
+        // Unwrap because it's not possible to construct a `Schema` that doesn't contain an
+        // application schema id.
+        view_id.unwrap()
+    }
```

Comment:
Two new ideas 🤣

1. Same, but less lines

```rust
 ​    ​pub​ ​fn​ ​view_id​(​&​self​) -> DocumentViewId { 
 ​        ​ match​ ​self​.id.​clone​() { 
 ​            SchemaId​::​Application​(_, view_id) ​=>​ ​view_id, 
 ​            _ ​=>​ ​panic("it is not possible to construct a schema ...")​, 
 ​        }
 ​    }
```

2. Introduce a `ApplicationSchemaId(Name, DocumentViewId)` type and have `SchemaId::Application (ApplicationSchemaId)`

```rust
     pub fn id(&self) -> SchemaId {
         SchemaId::Application(self.id.clone())
     }

 ​    ​pub​ ​fn​ ​view_id​(​&​self​) -> DocumentViewId { 
 ​        ​ self.id().clone().view_id()
 ​    }
```

Reply:
Oh this actually gives me another idea :D 

```rust
pub enum SchemaVersion {
    System(u8),
    Application(DocumentViewId)
}
```

Reply:
Yes, that's cool!

The second idea is nice because it gives you the type it should have, so no weird checks required 👍

Reply:
Please have a look at the new changes I made today. I think now all weird checks are resolved!

Reply:
> Please have a look at the new changes I made today. I think now all weird checks are resolved!

I like it 👍 it's tidy and nice!